BigImageViewer/src/main/java/com/github/piasy/biv/view/BigImageView.java

```diff
@@ -342,17 +349,33 @@ public void showImage(Uri uri) {
     }
 
     public void showImage(final Uri thumbnail, final Uri uri) {
+        showImage(thumbnail, uri, false);
+    }
+
+    public void showImage(final Uri thumbnail, final Uri uri, final boolean delayImage) {
         mThumbnail = thumbnail;
         mUri = uri;
 
         clearThumbnailAndProgressIndicator();
-        mImageLoader.loadImage(hashCode(), uri, mInternalCallback);
+
+        mDelayImage = delayImage;
+        if (mDelayImage) {
+            mImageLoader.loadImage(hashCode(), thumbnail, mInternalCallback);
```

Comment:
Could we prefetch the main image here? I mean calling `BigImageViewer.prefetch(uri)` here.

Reply:
Yes I guess we could do that, I'll try...

Reply:
Done, working!